csharp/src/Ice/ConnectionI.cs

```diff
@@ -1941,6 +1941,9 @@ private bool validate(int operation)
                     throw new MarshalException($"Received ValidateConnection message with unexpected size {size}.");
                 }
                 TraceUtil.traceRecv(_readStream, this, _logger, _traceLevels);
+
+                // Client connection starts sending heartbeats once it's received the ValidateConnection message.
+                _idleTimeoutTransceiver?.scheduleHeartbeat();
```

Comment:
Can _idleTimeoutTransceiver ever be null here?

Reply:
Yes, it's null when the idle timeout is <=0 (i.e., disabled)
https://github.com/zeroc-ice/ice/blob/54548fe4679da948d3067a78cea752eab5895373/csharp/src/Ice/ConnectionI.cs#L1410